src/spatialdata/_core/_deepcopy.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+from copy import deepcopy as _deepcopy
+from functools import singledispatch
+
+from anndata import AnnData
+from dask.array.core import Array as DaskArray
+from dask.array.core import from_array
+from dask.dataframe.core import DataFrame as DaskDataFrame
+from geopandas import GeoDataFrame
+from multiscale_spatial_image import MultiscaleSpatialImage
+from spatial_image import SpatialImage
+
+from spatialdata._core.spatialdata import SpatialData
+from spatialdata._utils import multiscale_spatial_image_from_data_tree
+from spatialdata.models._utils import SpatialElement
+from spatialdata.models.models import Image2DModel, Image3DModel, Labels2DModel, Labels3DModel, PointsModel, get_model
+
+
+@singledispatch
+def deepcopy(element: SpatialData | SpatialElement) -> SpatialData | SpatialElement:
+    """
+    Deepcopy a SpatialData or SpatialElement object.
+
+    Deepcopy will load the data in memory. Using this function for large Dask-backed objects is discouraged. In that
+    case, please save the SpatialData object to a different disk location and read it back again.
+
+    Parameters
+    ----------
+    element
+        The SpatialData or SpatialElement object to deepcopy
+
+    Returns
+    -------
+    A deepcopy of the SpatialData or SpatialElement object
+    """
+    raise RuntimeError(f"Wrong type for deepcopy: {type(element)}")
+
+
+# In the implementations below, when the data is loaded from Dask, we first use compute() and then we deepcopy the data.
+# This leads to double copying the data, but since we expect the data to be small, this is acceptable.
+@deepcopy.register(SpatialData)
+def _(sdata: SpatialData) -> SpatialData:
+    elements_dict = {}
+    for _, element_name, element in sdata.gen_elements():
+        elements_dict[element_name] = deepcopy(element)
+    return SpatialData.from_elements_dict(elements_dict)
+
+
+@deepcopy.register(SpatialImage)
+def _(element: SpatialImage) -> SpatialImage:
+    model = get_model(element)
+    if isinstance(element.data, DaskArray):
+        element = element.compute()
+    copied = model.parse(element.copy(deep=True))
+    element.data = from_array(element.data)
```

Comment:
```suggestion
```
This is not required as the element is only computed within the scope of the function. Outside it stays a dask array.